src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -285,52 +293,69 @@ that was used:
-vectors. Note that this helps as a workaround when `M` is a sparse matrix, since
-the `inv` function is not available for sparse matrices. In this case, either
-use the option `use_inv=false` or convert the type of `M` as in
-`linear_map(Matrix(M), P)`.
-
-Internally, this function operates at the level of the `AbstractPolyhedron`
+polyhedron is bounded, which we check.
+
+If the matrix is known to be invertible, the the option `check_invertibility`
+can be used to skip the invertibility test.
+If the matrix inverse is even known, it can be specified with the option
+`inverse`, in which case we ignore the other options and also the original
+matrix `M`.
+
+The algorithms `"division"` and `"inverse"` give control - in case `M` is
+invertible - on whether the full matrix inverse is computed or only the left
+division on the normal vectors is used.
+Note that this helps as a workaround when `M` is a sparse matrix (since the
+`inv` function is not available for sparse matrices).
+In this case, either use the algorithm `"division"` or convert `M` to a dense
+matrix (as in `linear_map(Matrix(M), P)`).
+
+Internally, this function operates on the level of the `AbstractPolyhedron`
 interface, but the actual algorithm uses dispatch on the concrete type of `P`,
 depending on the algorithm that is used:
 
--  `_linear_map_vrep(M, P)` if the vertex approach is used
--  `_linear_map_hrep(M, P, use_inv)` if the invertibility criterion is used
+- `_linear_map_vrep(M, P)` if the vertex approach is used
+- `_linear_map_hrep(M, P, use_inv)` if the invertibility criterion is used,
+  where `use_inv` is determined by the algorithm
 
-New subtypes of the interface should write their own `_linear_map_vrep`
+New subtypes of the interface should define their own `_linear_map_vrep`
 (resp. `_linear_map_hrep`) for special handling of the linear map; otherwise
-the fallback implementation for `AbstractPolyhedron` is used (see below).
+the fallback implementation for `AbstractPolyhedron` is used.
 """
 function linear_map(M::AbstractMatrix{N},
                     P::AbstractPolyhedron{N};
+                    algorithm::String=(issparse(M) ? "division" : "inverse"),
                     check_invertibility::Bool=true,
                     cond_tol::Number=DEFAULT_COND_TOL,
-                    use_inv::Bool=!issparse(M)
-                   ) where {N<:Real}
+                    inverse::Union{AbstractMatrix{N}, Nothing}=nothing) where {N<:Real}
+    # if `inverse` is specified, we ignore other inputs and do not check them
+    if inverse != nothing
+        return _linear_map_hrep(M, P, true; inverse=inverse)
+    end
+
+    @assert M != nothing "a matrix must be specified unless the inverse is " *
+        "specified as well"
     @assert dim(P) == size(M, 2) "a linear map of size $(size(M)) cannot be " *
         "applied to a set of dimension $(dim(P))"
+    supported_algorithms = ["vrep", "inverse", "division"]
```

Comment:
i would remove this `supported_algorithms` check and the associated assertion altogether.

otherwise, can we add this as a const outside this function (so we dont have to allocate a `supported_algorithms` vector for each call to this function).

Reply:
> i would remove this supported_algorithms definition and the associated assertion altogether.

Hm, okay.

> otherwise, can we add this as a const outside this function (so we dont have to allocate a supported_algorithms vector for each call to this function).

I had assumed that the compiler would do that for me, but I was wrong.

```julia
julia> function foo(x)
    supported_algorithms = ["vrep", "inverse", "division"]
    @assert x in supported_algorithms "supported algorithms: $supported_algorithms, but got $x"
    return true
end
foo (generic function with 1 method)

julia> @code_lowered foo(1)
CodeInfo(
1 ─      supported_algorithms = Base.vect("vrep", "inverse", "division")
│   %2 = x in supported_algorithms
└──      goto #3 if not %2
2 ─      goto #4
3 ─ %5 = Base.getproperty(Base.Main, :Base)
│   %6 = Base.getproperty(%5, :string)
│   %7 = Base.string("supported algorithms: ", supported_algorithms, ", but got ", x)
│   %8 = (%6)(%7)
│   %9 = Base.AssertionError(%8)
└──      Base.throw(%9)
4 ┄      return true
)
```